### PR TITLE
TandemFoil: physics stack + AdamW LR sweep (combine best of #2414 + #2416)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

Two orthogonal improvements have been independently validated on TandemFoil:
1. Physics features (PR #2414, tanjiro): TE coord frame + Cp panel + asinh pressure + residual prediction → val_mae=262.82 (new best, +2.4% vs baseline)
2. AdamW optimizer (PR #2416, alphonse): beats Lion by 38% on AirfRANS; on TandemFoil only 2 epochs were compared

These improvements should compound. This experiment combines BOTH and sweeps the AdamW LR to find the optimal setting. At model_slices=64 (vs default 96), we get more epochs per 30-minute budget (~4-5 epochs vs 2). This is the most important TandemFoil experiment in this round.

## Current Baseline

- **val_primary/surface_pressure_mae: 262.82** (PR #2414, tanjiro — physics features + Lion lr=3e-4)
- **test_primary/surface_pressure_mae: 257.51**

## Instructions

Run 4 parallel trials with `CUDA_VISIBLE_DEVICES`. Use `--wandb_group radford-tandem-physics-adamw` for all runs. All trials use the winning physics stack from #2414 plus AdamW at model_slices=64.

Physics flags for all trials: `--enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only --asinh-pressure --residual-prediction --enable-pressure-prior-addition`

### Trial 0 — GPU 0: AdamW lr=3e-4
```bash
CUDA_VISIBLE_DEVICES=0 python train.py --dataset tandemfoil --optimizer adamw --lr 3e-4 --cosine_t_max 30 --model-slices 64 --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only --asinh-pressure --residual-prediction --enable-pressure-prior-addition --wandb_group radford-tandem-physics-adamw --wandb_name tanjiro/tandem-phys-adamw-3e4 --agent tanjiro
```

### Trial 1 — GPU 1: AdamW lr=5e-4
```bash
CUDA_VISIBLE_DEVICES=1 python train.py --dataset tandemfoil --optimizer adamw --lr 5e-4 --cosine_t_max 30 --model-slices 64 --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only --asinh-pressure --residual-prediction --enable-pressure-prior-addition --wandb_group radford-tandem-physics-adamw --wandb_name tanjiro/tandem-phys-adamw-5e4 --agent tanjiro
```

### Trial 2 — GPU 2: AdamW lr=8e-4
```bash
CUDA_VISIBLE_DEVICES=2 python train.py --dataset tandemfoil --optimizer adamw --lr 8e-4 --cosine_t_max 30 --model-slices 64 --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only --asinh-pressure --residual-prediction --enable-pressure-prior-addition --wandb_group radford-tandem-physics-adamw --wandb_name tanjiro/tandem-phys-adamw-8e4 --agent tanjiro
```

### Trial 3 — GPU 3: AdamW lr=5e-4 at slices=96 (compare throughput)
```bash
CUDA_VISIBLE_DEVICES=3 python train.py --dataset tandemfoil --optimizer adamw --lr 5e-4 --cosine_t_max 30 --model-slices 96 --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only --asinh-pressure --residual-prediction --enable-pressure-prior-addition --wandb_group radford-tandem-physics-adamw --wandb_name tanjiro/tandem-phys-adamw-5e4-sl96 --agent tanjiro
```

Run all 4 in parallel (background each with `&`). Wait for all to finish.

**Critical notes:**
- The expected improvement path: physics (262.82) + AdamW should significantly beat baseline
- Report: epochs completed, val_primary/surface_pressure_mae for each trial (best across epochs)
- Also report the per-split val MAEs: val_single_in_dist, val_geom_camber_rc, val_geom_camber_cruise, val_re_rand
- Compare Trial 3 (slices=96, 2 epochs) vs Trial 1 (slices=64, ~4 epochs) at same lr=5e-4

## Results

_To be filled by student after runs complete._

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT tanjiro:

## Results

### Summary

Physics stack + AdamW compounds significantly. **Trial 3 (slices=96, AdamW lr=5e-4) is the clear winner** with a 12.1% val improvement and 11.9% test improvement over the baseline.

### All Trials

| Trial | LR | Slices | Epochs | val_primary/surface_pressure_mae | test_primary/surface_pressure_mae | W&B run |
|-------|-----|--------|--------|----------------------------------|-----------------------------------|---------|
| 0 | 3e-4 | 64 | 2 | 242.51 | 235.43 | tpmpyl1t |
| 1 | 5e-4 | 64 | 1 | 363.20 | 350.65 | hitriwza |
| 2 | 8e-4 | 64 | 2 | 254.01 | 247.07 | xt9jz1ld |
| **3** | **5e-4** | **96** | **2** | **231.04** | **226.78** | **v3hwb2xj** |
| Baseline (PR #2414) | Lion 3e-4 | 96 | 2 | 262.82 | 257.51 | 1zbp5dlu |

### Best Trial: Trial 3 (AdamW lr=5e-4, slices=96)

**val_primary/surface_pressure_mae: 231.04** (baseline: 262.82, **-12.1%**)
**test_primary/surface_pressure_mae: 226.78** (baseline: 257.51, **-11.9%**)

Per-split val MAE:
- val_single_in_dist: 273.54 (baseline: ~408 at epoch 1)
- val_geom_camber_rc: 244.22
- val_geom_camber_cruise: 202.83
- val_re_rand: 203.57

Per-split test MAE:
- test_single_in_dist: 240.32
- test_geom_camber_rc: 238.00
- test_geom_camber_cruise: 202.67
- test_re_rand: 226.12

**Command:**
```bash
CUDA_VISIBLE_DEVICES=3 python train.py --dataset tandemfoil --optimizer adamw --lr 5e-4 --cosine-t-max 30 --model-slices 96 --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only --asinh-pressure --residual-prediction --enable-pressure-prior-addition --wandb-group radford-tandem-physics-adamw --wandb-name tanjiro/tandem-phys-adamw-5e4-sl96 --agent tanjiro
```

**Peak memory:** Standard (no OOM issues on RTX PRO 6000 Blackwell)
**W&B run ID:** v3hwb2xj
**W&B group:** radford-tandem-physics-adamw

### Analysis

**What happened:**
- The hypothesis is strongly confirmed: physics features (TE coord frame + Cp panel + asinh pressure + residual prediction) and AdamW optimizer compound together for a massive improvement.
- All trials that completed 2 epochs beat the baseline — even lr=8e-4 (Trial 2) at slices=64 achieved 254.01 vs 262.82.
- **slices=96 > slices=64** at same lr=5e-4: Trial 3 (slices=96, 2 epochs) achieved 231.04 vs Trial 1 (slices=64, 1 epoch, 363.20). The PR hypothesis that slices=64 would enable 4-5 epochs was incorrect — with physics features, slices=64 still only gets 1-2 epochs. At 2 epochs each, the higher-resolution model wins decisively.
- **LR ranking at slices=64**: lr=3e-4 (242.51) > lr=8e-4 (254.01) > lr=5e-4 (363.20, but only 1 epoch). Lower LR is more stable.
- The cosine_t_max=30 creates a per-step oscillating schedule (30-step half-period, ~25 cycles per 750-step epoch). Despite this rapid oscillation, the model learns well — the LR returns to its initial value after every even epoch.
- Infinity values in test_geom_camber_cruise/mae_vol_p persist (known early-training EMA artifact from prior experiments).

**Note:** The PR commands used underscore-delimited flags (`--cosine_t_max`, `--wandb_group`, `--wandb_name`) which argparse rejected. Fixed to hyphens (`--cosine-t-max`, `--wandb-group`, `--wandb-name`).

### Suggested follow-ups

1. **Run Trial 3 config with more epochs** — with slices=96 at lr=5e-4, the model was still improving strongly at epoch 2. A longer timeout or checkpoint-resume could push this further.
2. **Try AdamW lr=3e-4 at slices=96** — lr=3e-4 was the best at slices=64, and slices=96 was the best resolution. Combining both could yield further gains.
3. **Cosine schedule tuning** — the per-step cosine_t_max=30 is quite aggressive. A longer T_max (e.g. 150 or epoch-aligned) might give smoother convergence.
4. **Weight decay sweep** — all trials used the default weight_decay. AdamW's decoupled weight decay is a key advantage; tuning it could help.
